### PR TITLE
salt: stop ping when all minions have answered

### PR DESCRIPTION
**Pull request by jan--f**
Signed-off-by: Jan Fajerski <jfajerski@suse.com>

### CODE REVIEW

File: teuthology/salt.py [hunk 1]
```diff
@@ -177,34 +180,25 @@ def start_minions(self):
...
-                log.debug("Attempt {n}/10 to ping Salt Minion {m}".format(
-                    m=mid,
-                    n=wl+1,
-                ))
+    def ping_minion(self, mid):
+        """Pings a minion, raises exception if it doesn't respond"""
+        self.__ping("sudo salt '{}' test.ping".format(mid), 1)
+
+    def ping_minions(self):
+        """Pings minions with this clusers job_id, raises exception if they don't respond"""
```
Comment (smithfarm) [inline teuthology/salt.py]: with this cluster's

File: teuthology/salt.py [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
 import time
 
 from os.path import isfile
 from netifaces import ifaddresses
 
 import teuthology
+from .contextutil import safe_while
 from .misc import sh
 from .orchestra import run
+from .suite import WaitException
```
Comment (smithfarm) [inline teuthology/salt.py]: why are you importing WaitException?

File: teuthology/task/ceph_cm_salt.py [hunk 3]
```diff
@@ -15,7 +15,8 @@ def task(ctx, config):
     salt = Salt(ctx, config)
     salt.init_minions()
     salt.start_minions()
-    salt.ping_minions_parallel()
+    salt.ping_minions()
+    salt.ping_minion(salt.minions[1])
```
Comment (smithfarm) [inline teuthology/task/ceph_cm_salt.py]: shouldn't this be `[0]` so it succeeds if there's only one remote?

Comment (jan--f) [inline teuthology/task/ceph_cm_salt.py]: right...this should probably just go out. Put it in there for testing

**Comment by jan--f** [comment]
@smithfarm feel free to merge

**Pull request merged**